packages/datadog-instrumentations/src/child_process.js

```diff
@@ -0,0 +1,176 @@
+'use strict'
+
+const util = require('util')
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const childProcessChannelStart = channel('datadog:child_process:execution:start')
+const childProcessChannelFinish = channel('datadog:child_process:execution:finish')
+const childProcessChannelError = channel('datadog:child_process:execution:error')
+
+// ignored exec method because it calls to execFile directly
+const execAsyncMethods = ['execFile', 'spawn']
+const execSyncMethods = ['execFileSync', 'spawnSync']
+
+const names = ['child_process', 'node:child_process']
+
+// child_process and node:child_process returns the same object instance, we only want to add hooks once
+let patched = false
+names.forEach(name => {
+  addHook({ name }, childProcess => {
+    if (!patched) {
+      patched = true
+      shimmer.massWrap(childProcess, execAsyncMethods, wrapChildProcessAsyncMethod())
+      shimmer.massWrap(childProcess, execSyncMethods, wrapChildProcessSyncMethod())
+      shimmer.wrap(childProcess, 'execSync', wrapChildProcessSyncMethod(true))
+    }
+
+    return childProcess
+  })
+})
+
+function wrapChildProcessSyncMethod (shell = false) {
+  return function wrapMethod (childProcessMethod) {
+    return function () {
+      if (!childProcessChannelStart.hasSubscribers || arguments.length === 0) {
```

Comment:
Are you sure you don't even want to create a trace when no argument is given ? First time i'm seeing this kind of check in the instrumentations. Is there prior art on this ?

Reply:
I suppose that is not the usual thing to do because the purpose is to inform of possible shell execution vulnerabilities not tracing child_process execution. Therefore if no arguments are passed what is the purpose of creating a child_process span with no relevant information?

Reply:
Aaaah right this is only creating a trace for the backend waf to analyze ? it's not meant to be actually used a APM traces ? if so you can resolve the comment :+1:

Reply:
That's right.